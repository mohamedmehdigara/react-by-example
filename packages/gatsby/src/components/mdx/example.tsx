import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
