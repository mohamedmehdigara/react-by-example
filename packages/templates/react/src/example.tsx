import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
