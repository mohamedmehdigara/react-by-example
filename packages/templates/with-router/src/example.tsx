import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
