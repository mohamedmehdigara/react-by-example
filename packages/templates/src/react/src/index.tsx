import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
