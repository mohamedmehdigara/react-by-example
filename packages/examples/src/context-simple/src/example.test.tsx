import { cleanup, render, RenderResult } from '@testing-library/react'
import React from 'react'
import example from './example'

describe('Context', () => {
  let component: RenderResult

  beforeEach(() => {
    component = render(<>{example}</>)
  })

  afterEach(() => {
    cleanup()
  })

  it('should snapshot', () => {
    expect(component.container).toMatchSnapshot()
  })
})
