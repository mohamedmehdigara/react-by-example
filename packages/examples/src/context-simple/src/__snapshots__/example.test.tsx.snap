// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context should snapshot 1`] = `
<div>
  <p>
    default value is used when there is no provider
  </p>
  <pre>
    using value: \`
    default
    \`
  </pre>
  <div
    class="box"
    style="border-color: #ff5555; border-style: solid;"
  >
    <pre>
      providing value: \`
      #ff5555
      \`
    </pre>
    <pre>
      using value: \`
      #ff5555
      \`
    </pre>
    <div
      class="box"
      style="border-color: #50fa7b; border-style: dashed;"
    >
      <p>
        value pierces component boundaries
      </p>
      <pre>
        using value: \`
        #ff5555
        \`
      </pre>
      <div
        class="box"
        style="border-color: #8be9fd; border-style: solid;"
      >
        <pre>
          providing value: \`
          #8be9fd
          \`
        </pre>
        <p>
          providers can be nested
        </p>
        <p>
          closest value is used
        </p>
        <pre>
          using value: \`
          #8be9fd
          \`
        </pre>
      </div>
    </div>
  </div>
</div>
`;
